dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -174,14 +187,33 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 
                         auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
 
+                        winrt::guid registeredClsid{ GUID_NULL };
+                        if (AppModel::Identity::IsPackagedProcess())
+                        {
+                            THROW_IF_FAILED(PushNotificationHelpers::GetComRegistrationFromRegistry(expectedPushServerArgs.data(), registeredClsid));
+                        }
+
                         wil::unique_cotaskmem_string processName;
                         THROW_IF_FAILED(GetCurrentProcessPath(processName));
-                        THROW_IF_FAILED(notificationPlatform->RegisterLongRunningActivatorWithClsid(processName.get(), GetComServerClsid()));
+                        THROW_IF_FAILED(notificationPlatform->RegisterLongRunningActivatorWithClsid(processName.get(), registeredClsid));
 
                         std::wstring toastAppId{ RetrieveNotificationAppId() };
                         THROW_IF_FAILED(notificationPlatform->AddToastRegistrationMapping(processName.get(), toastAppId.c_str()));
                     }
 
+                    // Need to register sink on new channel creation if handlers are registered
+                    bool registeredEvent{ false };
+                    {
+                        auto lock{ m_lock.lock_shared() };
+                        registeredEvent = bool(m_foregroundHandlers);
+                    }
+
+                    // Channel creation removes old sink, need to add sink if foreground handlers.
```

Comment:
nit: fix this comment
"Need to recreate the Foreground sinks since channel creation removes the old sink"